Server/Application/WalletMethods/Query/GetWalletQuery.cs

```diff
@@ -0,0 +1,35 @@
+﻿using MediatR;
+using Endava_Project.Server.Models;
+using System;
+using Endava_Project.Server.Data;
+using System.Threading.Tasks;
+using System.Threading;
+using Microsoft.EntityFrameworkCore;
+using System.Linq;
+
+namespace Endava_Project.Server.Application.WalletMethods.Query
+{
+    public class GetWalletQuery : IRequest<Wallet>
+    {
+        public string UserId { get; set; }
+        public Guid WalletId { get; set; }
+    }
+
+    public class GetWalletQueryHandler : IRequestHandler<GetWalletQuery, Wallet>
+    {
+        private readonly ApplicationDbContext context;
+
+        public GetWalletQueryHandler(ApplicationDbContext context)
+        {
+            this.context = context;
+        }
+
+        public async Task<Wallet> Handle(GetWalletQuery query, CancellationToken cancellationToken)
+        {
+            var userWallets = await context.Users.Include(x => x.Wallets).FirstOrDefaultAsync(x => x.Id == query.UserId);
```

Comment:
Instead of pulling the user with its wallets, you could instead retrieve the only wallet you're interested in.
Not entirely sure, but would something like this work?
```csharp
context.Users
    .Where(u => u.Id == query.UserId)
    .SelectMany(u => u.Wallets)
    .Where(w => w.Id = query.WalletId)
    .FirstOrDefault();
```

Reply:
Yes , but in this case we need to check if the wallet belongs to user , so a user can't recive data from a wallet which does not belong to him

Reply:
Yes, this query will return the wallet only if it belongs to the required user. No?

Reply:
Oh yes, is a better idea . Thank you . It works